src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -342,13 +343,28 @@ public static Translation2d[] getModuleTranslations() {
     };
   }
 
-  public Rotation2d getRotationToTranslation(Translation2d translation) {
-
+  public Rotation2d getRotationToTranslation(Pose2d translation) {
     double angle = Math.atan2(translation.getY() - pose.getY(), translation.getX() - pose.getX());
     return Rotation2d.fromRadians(angle);
   }
 
-  public Command pointTowardsTranslation(DoubleSupplier x, DoubleSupplier y) {
+  @AutoLogOutput(key = "Autoaim/Virtual Target")
+  public Pose2d getVirtualTarget() {
+
+    Pose2d target = FieldConstants.getSpeaker();
+
+    double distance = pose.minus(target).getTranslation().getNorm();
```

Comment:
account for time we move before shooting, maybe add lookahead distance as a parameter. also check if we need to recompute distance for the new virtual target to recompute tof